plugins/core-mail/core-mail.go

```diff
@@ -0,0 +1,63 @@
+package core_mail
+
+import (
+	"fmt"
+	"net/http"
+	"strconv"
+
+	blink_conn "github.com/blinkops/blink-sdk/plugin/connections"
+	log "github.com/sirupsen/logrus"
+	gomail "gopkg.in/mail.v2"
+)
+
+type Plugin struct{}
+
+func (p Plugin) HandleAuth(req *http.Request, conn map[string]string) error {
+	return nil
+}
+
+func (p Plugin) TestConnection(connection *blink_conn.ConnectionInstance) (bool, []byte) {
+	fromEmail, ok := connection.Data["email"]
+	if !ok {
+		return false, []byte("mail connection does not contain an email address")
+	}
+
+	password, ok := connection.Data["password"]
+	if !ok {
+		return false, []byte("mail connection does not contain a password")
+	}
+
+	smtpHost, ok := connection.Data["smtpHost"]
+	if !ok {
+		return false, []byte("mail connection does not contain smtp host server")
+	}
+	smtpPort, ok := connection.Data["smtpPort"]
+	if !ok {
+		return false, []byte("mail connection does not contain smtp host port")
+	}
+
+	port, err := strconv.Atoi(smtpPort)
+	if err != nil {
+		err = fmt.Errorf("provided smtp port is invalid, error: %v", err)
+		log.Error(err)
+		return false, []byte("test connection failed" + err.Error())
+	}
```

Comment:
There's no space between failed and provided
```suggestion
	if err != nil {
		return false, []byte("test connection failed: provided smtp port is invalid: " + err.Error())
	}
```